### Importing necessary modules

In [25]:
from keras.models import Sequential
from keras.layers import Dense
from keras.constraints import nonneg
import numpy as np
import pandas as pd
import nltk
import operator

### Dataset
Here Wikipedia Movie Plots dataset have been used.<br>
Dataset link: https://www.kaggle.com/jrobischon/wikipedia-movie-plots
* You can fit your own dataset

In [26]:
#data 
df = pd.read_csv('dataset/wiki_movie_plots_deduped.csv') #get it from Kaggle
STOP_WORDS = set(nltk.corpus.stopwords.words('english')) 
df.Plot.head()

0    A bartender is working at a saloon, serving dr...
1    The moon, painted with a smiling face hangs ov...
2    The film, just over a minute long, is composed...
3    Lasting just 61 seconds and consisting of two ...
4    The earliest known adaptation of the classic f...
Name: Plot, dtype: object

In [27]:
df.Plot[0]

"A bartender is working at a saloon, serving drinks to customers. After he fills a stereotypically Irish man's bucket with beer, Carrie Nation and her followers burst inside. They assault the Irish man, pulling his hat over his eyes and then dumping the beer over his head. The group then begin wrecking the bar, smashing the fixtures, mirrors, and breaking the cash register. The bartender then sprays seltzer water in Nation's face before a group of policemen appear and order everybody to leave.[1]"

### Necessary Functions

In [28]:
#filtering and tokenizing function
def sentence_to_wordlist(sentence, filters="!\"#$%&()*+,-./:;<=>?@[\\]^_`{|}~\t\n?,।!‍.'0123456789০১২৩৪৫৬৭৮৯‘\u200c–“”…‘"):
    translate_dict = dict((c, ' ') for c in filters)
    translate_map = str.maketrans(translate_dict)
    wordlist = sentence.translate(translate_map).split()
    return list(filter(lambda x: x not in STOP_WORDS, wordlist))

#function for calculating word similarity
def cosine_similarity(u, v):

    '''Cosine similarity reflects the degree of similariy between u and v'''
    
    distance = 0.0
    
    ### START CODE HERE ###
    # Compute the dot product between u and v (≈1 line)
    dot = np.dot(u,v)
    # Compute the L2 norm of u (≈1 line)
    norm_u = np.sqrt(np.sum(np.power(u,2)))
    
    # Compute the L2 norm of v (≈1 line)
    norm_v = np.sqrt(np.sum(np.power(v,2)))
    # Compute the cosine similarity defined by formula (1) (≈1 line)
    cosine_similarity = np.divide(dot,norm_u*norm_v)
    ### END CODE HERE ###
    
    return cosine_similarity

#determine simularity between two words
def word_similarity(word1,word2,weight,word_to_index,print_vec=False):
    e1 = weight[word_to_index[word1]]
    e2 = weight[word_to_index[word2]]
    if print_vec:
        print(word1,e1)
        print(word2,e2)
    return cosine_similarity(e1,e2)

#retruns most similiar words
def most_similiar(word,weight,word_to_index,count=5,print_vec=False):
    e1 = weight[word_to_index[word]]
    similarity = {}
    for v in vocab:
        if v is not word:
            e2 = weight[word_to_index[v]]
            similarity[v]=cosine_similarity(e1,e2)
            
    if print_vec:
        print(word,e)
        
    sorted_similarity = sorted(similarity.items(), key=operator.itemgetter(1),reverse=True)      
    return sorted_similarity[1:count+1]

#function for vocab and indexing
def word_idx_generate(text):
    '''
    --build vocabulary of unique words
    --create dictionary for 'indext to word' and word to index'
    '''
    # build vocabulary of unique words
    vocab = []
    for sent in text:
        for i in sent:
            if i not in vocab:
                vocab.append(i)
                
    word2id = {}
    id2word = {}
    for i,word in enumerate(vocab):
        word2id[word] = i
        id2word[i] = word
    return vocab,word2id,id2word

#dataset: CBOW Represantation
def cbow(text,window=4):
    '''Return dataset as one-hot encoded context(X) vs target(Y) words'''
    X = []
    Y = []
    X_context = []
    Y_target = []
    V = len(vocab)
    for sent in text:
        border = len(sent)
        end = window -1
        start = 0
        if len(sent)>=window:
            while (border>end):
                x_temp = []
                x_word = [] #op
                for i in range(start,end):
                    index = word_to_index[sent[i]]
                    temp = np.zeros((1,V))
                    temp[0][index] = 1
                    x_temp.append(temp)
                    x_word.append(sent[i]) #op
                    #print(i,sent[i],end,border) #op
                index = word_to_index[sent[end]]
                temp = np.zeros((1,V))
                temp[0][index] = 1
                Y.append(temp)
                X.append(np.sum(x_temp,axis=0)) #summping context words
                X_context.append(x_word)
                Y_target.append(sent[end]) #op
                start +=1
                end +=1
    
    return X,Y,x_context,y_target

#dataset: Skip-gram Represantation
def skip_gram(text,word_to_index,window=4):
    '''Return dataset as one-hot encoded context(X) vs target(Y) words'''
    X = []
    Y = []
    X_context = []
    Y_target = []
    V = len(vocab)
    for sent in text:
        border = len(sent)
        end = window-1
        start = 0
        if len(sent)>=window:
            while (border>end):
                index = word_to_index[sent[start]]
                temp = np.zeros((1,V))
                temp[0][index] = 1
                X.append(temp)
                y_temp = []
                y_word = []
                X_context.append(sent[start]) #op
                for i in range(start+1,end+1):
                    index = word_to_index[sent[i]]
                    temp = np.zeros((1,V))
                    temp[0][index] = 1
                    y_temp.append(temp)
                    y_word.append(sent[i])

                Y.append(np.sum(y_temp,axis=0)) #summping context words
                Y_target.append(y_word) #op
                start +=1
                end +=1
       
    X = np.array(X)
    Y = np.array(Y)

    return X,Y,X_context,Y_target

def contex_target(X_context,Y_target):
    '''prints context vs target words'''
    for i,x in enumerate(X_context):
        print(i,"Context :",x,'---> Target:',Y_target[i])

### Preparing Dataset for word vectors
We are taking first 200 plots for reducing time it takes

In [29]:
text = df[:200] #selecting first 200 entry
text = [sentence_to_wordlist(i.lower()) for i in text.Plot] #filtering and tokenizing plot text

In [52]:
text

[['bartender',
  'working',
  'saloon',
  'serving',
  'drinks',
  'customers',
  'fills',
  'stereotypically',
  'irish',
  'man',
  'bucket',
  'beer',
  'carrie',
  'nation',
  'followers',
  'burst',
  'inside',
  'assault',
  'irish',
  'man',
  'pulling',
  'hat',
  'eyes',
  'dumping',
  'beer',
  'head',
  'group',
  'begin',
  'wrecking',
  'bar',
  'smashing',
  'fixtures',
  'mirrors',
  'breaking',
  'cash',
  'register',
  'bartender',
  'sprays',
  'seltzer',
  'water',
  'nation',
  'face',
  'group',
  'policemen',
  'appear',
  'order',
  'everybody',
  'leave'],
 ['moon',
  'painted',
  'smiling',
  'face',
  'hangs',
  'park',
  'night',
  'young',
  'couple',
  'walking',
  'past',
  'fence',
  'learn',
  'railing',
  'look',
  'moon',
  'smiles',
  'embrace',
  'moon',
  'smile',
  'gets',
  'bigger',
  'sit',
  'bench',
  'tree',
  'moon',
  'view',
  'blocked',
  'causing',
  'frown',
  'last',
  'scene',
  'man',
  'fans',
  'woman',
  'hat',
  'moon',
  'left',

In [30]:
vocab, word2idx,idx2word = word_idx_generate(text)

In [31]:
X,Y,X_context,Y_target = skip_gram(text,word2idx)
print('X:',X.shape,'Y:',Y.shape)

X: (18360, 1, 5779) Y: (18360, 1, 5779)


Here we have followed skip-gram model. Shape of X and Y shows, we have **18360 context word against 18360 target word sets.** <br>
Here we can visualize **context word** Vs **target word** with *context_target()* function.

In [32]:
contex_target(X_context,Y_target)

0 Context : bartender ---> Target: ['working', 'saloon', 'serving']
1 Context : working ---> Target: ['saloon', 'serving', 'drinks']
2 Context : saloon ---> Target: ['serving', 'drinks', 'customers']
3 Context : serving ---> Target: ['drinks', 'customers', 'fills']
4 Context : drinks ---> Target: ['customers', 'fills', 'stereotypically']
5 Context : customers ---> Target: ['fills', 'stereotypically', 'irish']
6 Context : fills ---> Target: ['stereotypically', 'irish', 'man']
7 Context : stereotypically ---> Target: ['irish', 'man', 'bucket']
8 Context : irish ---> Target: ['man', 'bucket', 'beer']
9 Context : man ---> Target: ['bucket', 'beer', 'carrie']
10 Context : bucket ---> Target: ['beer', 'carrie', 'nation']
11 Context : beer ---> Target: ['carrie', 'nation', 'followers']
12 Context : carrie ---> Target: ['nation', 'followers', 'burst']
13 Context : nation ---> Target: ['followers', 'burst', 'inside']
14 Context : followers ---> Target: ['burst', 'inside', 'assault']
15 Context 

277 Context : large ---> Target: ['white', 'rabbit', 'rabbit']
278 Context : white ---> Target: ['rabbit', 'rabbit', 'hole']
279 Context : rabbit ---> Target: ['rabbit', 'hole', 'finds']
280 Context : rabbit ---> Target: ['hole', 'finds', 'tiny']
281 Context : hole ---> Target: ['finds', 'tiny', 'door']
282 Context : finds ---> Target: ['tiny', 'door', 'finds']
283 Context : tiny ---> Target: ['door', 'finds', 'bottle']
284 Context : door ---> Target: ['finds', 'bottle', 'labeled']
285 Context : finds ---> Target: ['bottle', 'labeled', 'drink']
286 Context : bottle ---> Target: ['labeled', 'drink', 'shrinks']
287 Context : labeled ---> Target: ['drink', 'shrinks', 'enough']
288 Context : drink ---> Target: ['shrinks', 'enough', 'pass']
289 Context : shrinks ---> Target: ['enough', 'pass', 'door']
290 Context : enough ---> Target: ['pass', 'door', 'eats']
291 Context : pass ---> Target: ['door', 'eats', 'something']
292 Context : door ---> Target: ['eats', 'something', 'labeled']
293 Co

477 Context : restores ---> Target: ['consciousness', 'dousing', 'water']
478 Context : consciousness ---> Target: ['dousing', 'water', 'comic']
479 Context : dousing ---> Target: ['water', 'comic', 'relief']
480 Context : water ---> Target: ['comic', 'relief', 'dance']
481 Context : comic ---> Target: ['relief', 'dance', 'hall']
482 Context : relief ---> Target: ['dance', 'hall', 'eastern']
483 Context : dance ---> Target: ['hall', 'eastern', 'stranger']
484 Context : hall ---> Target: ['eastern', 'stranger', 'forced']
485 Context : eastern ---> Target: ['stranger', 'forced', 'dance']
486 Context : stranger ---> Target: ['forced', 'dance', 'locals']
487 Context : forced ---> Target: ['dance', 'locals', 'fire']
488 Context : dance ---> Target: ['locals', 'fire', 'feet']
489 Context : locals ---> Target: ['fire', 'feet', 'door']
490 Context : fire ---> Target: ['feet', 'door', 'suddenly']
491 Context : feet ---> Target: ['door', 'suddenly', 'opens']
492 Context : door ---> Target: ['sud

876 Context : fight ---> Target: ['santa', 'claus', 'leaves']
877 Context : santa ---> Target: ['claus', 'leaves', 'home']
878 Context : claus ---> Target: ['leaves', 'home', 'sleigh']
879 Context : leaves ---> Target: ['home', 'sleigh', 'reindeer']
880 Context : home ---> Target: ['sleigh', 'reindeer', 'enters']
881 Context : sleigh ---> Target: ['reindeer', 'enters', 'children']
882 Context : reindeer ---> Target: ['enters', 'children', 'house']
883 Context : enters ---> Target: ['children', 'house', 'chimney']
884 Context : children ---> Target: ['house', 'chimney', 'leaves']
885 Context : house ---> Target: ['chimney', 'leaves', 'presents']
886 Context : chimney ---> Target: ['leaves', 'presents', 'children']
887 Context : leaves ---> Target: ['presents', 'children', 'come']
888 Context : presents ---> Target: ['children', 'come', 'stairs']
889 Context : children ---> Target: ['come', 'stairs', 'enjoy']
890 Context : come ---> Target: ['stairs', 'enjoy', 'presents']
891 Context : r

1276 Context : fire ---> Target: ['thug', 'rescuer', 'fight']
1277 Context : thug ---> Target: ['rescuer', 'fight', 'roof']
1278 Context : rescuer ---> Target: ['fight', 'roof', 'house']
1279 Context : young ---> Target: ['couple', 'decides', 'elope']
1280 Context : couple ---> Target: ['decides', 'elope', 'caught']
1281 Context : decides ---> Target: ['elope', 'caught', 'midst']
1282 Context : elope ---> Target: ['caught', 'midst', 'romantic']
1283 Context : caught ---> Target: ['midst', 'romantic', 'moment']
1284 Context : midst ---> Target: ['romantic', 'moment', 'woman']
1285 Context : romantic ---> Target: ['moment', 'woman', 'angry']
1286 Context : moment ---> Target: ['woman', 'angry', 'father']
1287 Context : woman ---> Target: ['angry', 'father', 'make']
1288 Context : angry ---> Target: ['father', 'make', 'plans']
1289 Context : father ---> Target: ['make', 'plans', 'leave']
1290 Context : make ---> Target: ['plans', 'leave', 'thief']
1291 Context : plans ---> Target: ['leave

1617 Context : borrows ---> Target: ['coin', 'win', 'fortune']
1618 Context : coin ---> Target: ['win', 'fortune', 'girl']
1619 Context : win ---> Target: ['fortune', 'girl', 'wakes']
1620 Context : fortune ---> Target: ['girl', 'wakes', 'two']
1621 Context : girl ---> Target: ['wakes', 'two', 'miss']
1622 Context : wakes ---> Target: ['two', 'miss', 'wandering']
1623 Context : two ---> Target: ['miss', 'wandering', 'streets']
1624 Context : miss ---> Target: ['wandering', 'streets', 'gambler']
1625 Context : wandering ---> Target: ['streets', 'gambler', 'finds']
1626 Context : streets ---> Target: ['gambler', 'finds', 'girl']
1627 Context : gambler ---> Target: ['finds', 'girl', 'dead']
1628 Context : story ---> Target: ['told', 'moving', 'picture']
1629 Context : told ---> Target: ['moving', 'picture', 'world']
1630 Context : moving ---> Target: ['picture', 'world', 'reads']
1631 Context : mary ---> Target: ['pickford', 'plays', 'priscilla']
1632 Context : pickford ---> Target: ['pla

1876 Context : descend ---> Target: ['pocahontas', 'favorite', 'daughter']
1877 Context : pocahontas ---> Target: ['favorite', 'daughter', 'king']
1878 Context : favorite ---> Target: ['daughter', 'king', 'throws']
1879 Context : daughter ---> Target: ['king', 'throws', 'father']
1880 Context : king ---> Target: ['throws', 'father', 'begs']
1881 Context : throws ---> Target: ['father', 'begs', 'fervently']
1882 Context : father ---> Target: ['begs', 'fervently', 'white']
1883 Context : begs ---> Target: ['fervently', 'white', 'man']
1884 Context : fervently ---> Target: ['white', 'man', 'life']
1885 Context : white ---> Target: ['man', 'life', 'spared']
1886 Context : man ---> Target: ['life', 'spared', 'powhatan']
1887 Context : life ---> Target: ['spared', 'powhatan', 'relents']
1888 Context : spared ---> Target: ['powhatan', 'relents', 'orders']
1889 Context : powhatan ---> Target: ['relents', 'orders', 'release']
1890 Context : relents ---> Target: ['orders', 'release', 'captain']


2176 Context : fortune ---> Target: ['read', 'slip', 'away']
2177 Context : read ---> Target: ['slip', 'away', 'hand']
2178 Context : slip ---> Target: ['away', 'hand', 'hand']
2179 Context : away ---> Target: ['hand', 'hand', 'rough']
2180 Context : hand ---> Target: ['hand', 'rough', 'hewn']
2181 Context : hand ---> Target: ['rough', 'hewn', 'fence']
2182 Context : rough ---> Target: ['hewn', 'fence', 'field']
2183 Context : hewn ---> Target: ['fence', 'field', 'daisies']
2184 Context : fence ---> Target: ['field', 'daisies', 'town']
2185 Context : field ---> Target: ['daisies', 'town', 'fortuneteller']
2186 Context : daisies ---> Target: ['town', 'fortuneteller', 'examining']
2187 Context : town ---> Target: ['fortuneteller', 'examining', 'millie']
2188 Context : fortuneteller ---> Target: ['examining', 'millie', 'palm']
2189 Context : examining ---> Target: ['millie', 'palm', 'martha']
2190 Context : millie ---> Target: ['palm', 'martha', 'distracted']
2191 Context : palm ---> Targ

2476 Context : tin ---> Target: ['woodsman', 'oil', 'find']
2477 Context : woodsman ---> Target: ['oil', 'find', 'eureka']
2478 Context : oil ---> Target: ['find', 'eureka', 'cat']
2479 Context : find ---> Target: ['eureka', 'cat', 'enter']
2480 Context : eureka ---> Target: ['cat', 'enter', 'forest']
2481 Context : cat ---> Target: ['enter', 'forest', 'momba']
2482 Context : enter ---> Target: ['forest', 'momba', 'witch']
2483 Context : forest ---> Target: ['momba', 'witch', 'flies']
2484 Context : momba ---> Target: ['witch', 'flies', 'window']
2485 Context : witch ---> Target: ['flies', 'window', 'soldiers']
2486 Context : flies ---> Target: ['window', 'soldiers', 'come']
2487 Context : window ---> Target: ['soldiers', 'come', 'cottage']
2488 Context : soldiers ---> Target: ['come', 'cottage', 'captured']
2489 Context : come ---> Target: ['cottage', 'captured', 'led']
2490 Context : cottage ---> Target: ['captured', 'led', 'witch']
2491 Context : captured ---> Target: ['led', 'witch

2776 Context : follow ---> Target: ['sir', 'daniel', 'moat']
2777 Context : sir ---> Target: ['daniel', 'moat', 'house']
2778 Context : daniel ---> Target: ['moat', 'house', 'dick']
2779 Context : moat ---> Target: ['house', 'dick', 'confirms']
2780 Context : house ---> Target: ['dick', 'confirms', 'sir']
2781 Context : dick ---> Target: ['confirms', 'sir', 'daniel']
2782 Context : confirms ---> Target: ['sir', 'daniel', 'murderer']
2783 Context : sir ---> Target: ['daniel', 'murderer', 'father']
2784 Context : daniel ---> Target: ['murderer', 'father', 'escapes']
2785 Context : murderer ---> Target: ['father', 'escapes', 'injured']
2786 Context : father ---> Target: ['escapes', 'injured', 'moat']
2787 Context : escapes ---> Target: ['injured', 'moat', 'house']
2788 Context : injured ---> Target: ['moat', 'house', 'rescued']
2789 Context : moat ---> Target: ['house', 'rescued', 'outlaws']
2790 Context : house ---> Target: ['rescued', 'outlaws', 'black']
2791 Context : rescued ---> Targ

3076 Context : provide ---> Target: ['captain', 'arblaster', 'outlaw']
3077 Context : captain ---> Target: ['arblaster', 'outlaw', 'lawless']
3078 Context : arblaster ---> Target: ['outlaw', 'lawless', 'pensioning']
3079 Context : outlaw ---> Target: ['lawless', 'pensioning', 'settling']
3080 Context : lawless ---> Target: ['pensioning', 'settling', 'tunstall']
3081 Context : pensioning ---> Target: ['settling', 'tunstall', 'hamlet']
3082 Context : settling ---> Target: ['tunstall', 'hamlet', 'lawless']
3083 Context : tunstall ---> Target: ['hamlet', 'lawless', 'volte']
3084 Context : hamlet ---> Target: ['lawless', 'volte', 'face']
3085 Context : lawless ---> Target: ['volte', 'face', 'returning']
3086 Context : volte ---> Target: ['face', 'returning', 'franciscan']
3087 Context : face ---> Target: ['returning', 'franciscan', 'order']
3088 Context : returning ---> Target: ['franciscan', 'order', 'taking']
3089 Context : franciscan ---> Target: ['order', 'taking', 'name']
3090 Context 

3376 Context : men ---> Target: ['arrested', 'conspiracy', 'government']
3377 Context : arrested ---> Target: ['conspiracy', 'government', 'jack']
3378 Context : conspiracy ---> Target: ['government', 'jack', 'imprisoned']
3379 Context : government ---> Target: ['jack', 'imprisoned', 'one']
3380 Context : jack ---> Target: ['imprisoned', 'one', 'conspirators']
3381 Context : imprisoned ---> Target: ['one', 'conspirators', 'jail']
3382 Context : one ---> Target: ['conspirators', 'jail', 'jack']
3383 Context : conspirators ---> Target: ['jail', 'jack', 'tries']
3384 Context : jail ---> Target: ['jack', 'tries', 'make']
3385 Context : jack ---> Target: ['tries', 'make', 'escape']
3386 Context : tries ---> Target: ['make', 'escape', 'throws']
3387 Context : make ---> Target: ['escape', 'throws', 'guard']
3388 Context : escape ---> Target: ['throws', 'guard', 'ground']
3389 Context : throws ---> Target: ['guard', 'ground', 'sooner']
3390 Context : guard ---> Target: ['ground', 'sooner', 'le

3676 Context : york ---> Target: ['city', 'husband', 'works']
3677 Context : city ---> Target: ['husband', 'works', 'musician']
3678 Context : husband ---> Target: ['works', 'musician', 'must']
3679 Context : works ---> Target: ['musician', 'must', 'often']
3680 Context : musician ---> Target: ['must', 'often', 'travel']
3681 Context : must ---> Target: ['often', 'travel', 'work']
3682 Context : often ---> Target: ['travel', 'work', 'returning']
3683 Context : travel ---> Target: ['work', 'returning', 'wallet']
3684 Context : work ---> Target: ['returning', 'wallet', 'taken']
3685 Context : returning ---> Target: ['wallet', 'taken', 'gangster']
3686 Context : wallet ---> Target: ['taken', 'gangster', 'wife']
3687 Context : taken ---> Target: ['gangster', 'wife', 'goes']
3688 Context : gangster ---> Target: ['wife', 'goes', 'ball']
3689 Context : wife ---> Target: ['goes', 'ball', 'man']
3690 Context : goes ---> Target: ['ball', 'man', 'tries']
3691 Context : ball ---> Target: ['man', '

3975 Context : ship ---> Target: ['von', 'kammacher', 'learns']
3976 Context : von ---> Target: ['kammacher', 'learns', 'ingigerd']
3977 Context : kammacher ---> Target: ['learns', 'ingigerd', 'boyfriend']
3978 Context : learns ---> Target: ['ingigerd', 'boyfriend', 'thus']
3979 Context : ingigerd ---> Target: ['boyfriend', 'thus', 'backs']
3980 Context : boyfriend ---> Target: ['thus', 'backs', 'shortly']
3981 Context : thus ---> Target: ['backs', 'shortly', 'called']
3982 Context : backs ---> Target: ['shortly', 'called', 'treat']
3983 Context : shortly ---> Target: ['called', 'treat', 'young']
3984 Context : called ---> Target: ['treat', 'young', 'russian']
3985 Context : treat ---> Target: ['young', 'russian', 'girl']
3986 Context : young ---> Target: ['russian', 'girl', 'seasickness']
3987 Context : russian ---> Target: ['girl', 'seasickness', 'nearly']
3988 Context : girl ---> Target: ['seasickness', 'nearly', 'get']
3989 Context : seasickness ---> Target: ['nearly', 'get', 'roma

4175 Context : health ---> Target: ['relationship', 'blossoms', 'happiness']
4176 Context : relationship ---> Target: ['blossoms', 'happiness', 'returns']
4177 Context : blossoms ---> Target: ['happiness', 'returns', 'friedrich']
4178 Context : happiness ---> Target: ['returns', 'friedrich', 'life']
4179 Context : returns ---> Target: ['friedrich', 'life', 'realizes']
4180 Context : friedrich ---> Target: ['life', 'realizes', 'eva']
4181 Context : life ---> Target: ['realizes', 'eva', 'good']
4182 Context : realizes ---> Target: ['eva', 'good', 'mother']
4183 Context : eva ---> Target: ['good', 'mother', 'children']
4184 Context : barney ---> Target: ['oldfield', 'races', 'speeding']
4185 Context : oldfield ---> Target: ['races', 'speeding', 'locomotive']
4186 Context : races ---> Target: ['speeding', 'locomotive', 'rescue']
4187 Context : speeding ---> Target: ['locomotive', 'rescue', 'damsel']
4188 Context : locomotive ---> Target: ['rescue', 'damsel', 'distress']
4189 Context : resc

4475 Context : schoolmaster ---> Target: ['john', 'hart', 'played']
4476 Context : john ---> Target: ['hart', 'played', 'ray']
4477 Context : hart ---> Target: ['played', 'ray', 'local']
4478 Context : played ---> Target: ['ray', 'local', 'minister']
4479 Context : ray ---> Target: ['local', 'minister', 'rev']
4480 Context : local ---> Target: ['minister', 'rev', 'cole']
4481 Context : minister ---> Target: ['rev', 'cole', 'played']
4482 Context : rev ---> Target: ['cole', 'played', 'taylor']
4483 Context : cole ---> Target: ['played', 'taylor', 'calls']
4484 Context : played ---> Target: ['taylor', 'calls', 'cabin']
4485 Context : taylor ---> Target: ['calls', 'cabin', 'flowers']
4486 Context : calls ---> Target: ['cabin', 'flowers', 'unwelcome']
4487 Context : cabin ---> Target: ['flowers', 'unwelcome', 'suitor']
4488 Context : flowers ---> Target: ['unwelcome', 'suitor', 'revenge']
4489 Context : unwelcome ---> Target: ['suitor', 'revenge', 'blue']
4490 Context : suitor ---> Target:

4775 Context : blanche ---> Target: ['sweet', 'prevented', 'uncle']
4776 Context : sweet ---> Target: ['prevented', 'uncle', 'spottiswoode']
4777 Context : prevented ---> Target: ['uncle', 'spottiswoode', 'aitken']
4778 Context : uncle ---> Target: ['spottiswoode', 'aitken', 'pursuing']
4779 Context : spottiswoode ---> Target: ['aitken', 'pursuing', 'tormented']
4780 Context : aitken ---> Target: ['pursuing', 'tormented', 'visions']
4781 Context : pursuing ---> Target: ['tormented', 'visions', 'death']
4782 Context : tormented ---> Target: ['visions', 'death', 'suffering']
4783 Context : visions ---> Target: ['death', 'suffering', 'deciding']
4784 Context : death ---> Target: ['suffering', 'deciding', 'murder']
4785 Context : suffering ---> Target: ['deciding', 'murder', 'way']
4786 Context : deciding ---> Target: ['murder', 'way', 'things']
4787 Context : murder ---> Target: ['way', 'things', 'young']
4788 Context : way ---> Target: ['things', 'young', 'man']
4789 Context : things ---

5075 Context : must ---> Target: ['used', 'personal', 'expenditures']
5076 Context : used ---> Target: ['personal', 'expenditures', 'monty']
5077 Context : personal ---> Target: ['expenditures', 'monty', 'spends']
5078 Context : expenditures ---> Target: ['monty', 'spends', 'lavishly']
5079 Context : monty ---> Target: ['spends', 'lavishly', 'invests']
5080 Context : spends ---> Target: ['lavishly', 'invests', 'stock']
5081 Context : lavishly ---> Target: ['invests', 'stock', 'makes']
5082 Context : invests ---> Target: ['stock', 'makes', 'bet']
5083 Context : stock ---> Target: ['makes', 'bet', 'prize']
5084 Context : makes ---> Target: ['bet', 'prize', 'fight']
5085 Context : bet ---> Target: ['prize', 'fight', 'bet']
5086 Context : prize ---> Target: ['fight', 'bet', 'stocks']
5087 Context : fight ---> Target: ['bet', 'stocks', 'pay']
5088 Context : bet ---> Target: ['stocks', 'pay', 'desperation']
5089 Context : stocks ---> Target: ['pay', 'desperation', 'rents']
5090 Context : pay

5375 Context : state ---> Target: ['man', 'boots', 'charlie']
5376 Context : man ---> Target: ['boots', 'charlie', 'tries']
5377 Context : boots ---> Target: ['charlie', 'tries', 'another']
5378 Context : charlie ---> Target: ['tries', 'another', 'room']
5379 Context : tries ---> Target: ['another', 'room', 'key']
5380 Context : another ---> Target: ['room', 'key', 'gets']
5381 Context : room ---> Target: ['key', 'gets', 'starts']
5382 Context : key ---> Target: ['gets', 'starts', 'undress']
5383 Context : gets ---> Target: ['starts', 'undress', 'goes']
5384 Context : starts ---> Target: ['undress', 'goes', 'bed']
5385 Context : undress ---> Target: ['goes', 'bed', 'meanwhile']
5386 Context : goes ---> Target: ['bed', 'meanwhile', 'man']
5387 Context : bed ---> Target: ['meanwhile', 'man', 'across']
5388 Context : meanwhile ---> Target: ['man', 'across', 'hall']
5389 Context : man ---> Target: ['across', 'hall', 'leaves']
5390 Context : across ---> Target: ['hall', 'leaves', 'wife']
53

5675 Context : home ---> Target: ['becomes', 'clear', 'royal']
5676 Context : becomes ---> Target: ['clear', 'royal', 'heralds']
5677 Context : clear ---> Target: ['royal', 'heralds', 'every']
5678 Context : royal ---> Target: ['heralds', 'every', 'woman']
5679 Context : heralds ---> Target: ['every', 'woman', 'town']
5680 Context : every ---> Target: ['woman', 'town', 'tried']
5681 Context : woman ---> Target: ['town', 'tried', 'failed']
5682 Context : town ---> Target: ['tried', 'failed', 'wear']
5683 Context : tried ---> Target: ['failed', 'wear', 'slippers']
5684 Context : failed ---> Target: ['wear', 'slippers', 'except']
5685 Context : wear ---> Target: ['slippers', 'except', 'cinderella']
5686 Context : slippers ---> Target: ['except', 'cinderella', 'prince']
5687 Context : except ---> Target: ['cinderella', 'prince', 'charming']
5688 Context : cinderella ---> Target: ['prince', 'charming', 'immediately']
5689 Context : prince ---> Target: ['charming', 'immediately', 'goes']
569

6075 Context : describe ---> Target: ['someone', 'regularly', 'loitered']
6076 Context : someone ---> Target: ['regularly', 'loitered', 'near']
6077 Context : regularly ---> Target: ['loitered', 'near', 'actors']
6078 Context : loitered ---> Target: ['near', 'actors', 'entrances']
6079 Context : near ---> Target: ['actors', 'entrances', 'theaters']
6080 Context : actors ---> Target: ['entrances', 'theaters', 'hoping']
6081 Context : entrances ---> Target: ['theaters', 'hoping', 'meet']
6082 Context : theaters ---> Target: ['hoping', 'meet', 'players']
6083 Context : hoping ---> Target: ['meet', 'players', 'perhaps']
6084 Context : meet ---> Target: ['players', 'perhaps', 'land']
6085 Context : players ---> Target: ['perhaps', 'land', 'job']
6086 Context : perhaps ---> Target: ['land', 'job', 'onstage']
6087 Context : land ---> Target: ['job', 'onstage', 'backstage']
6088 Context : film ---> Target: ['lillian', 'travers', 'wealthy']
6089 Context : lillian ---> Target: ['travers', 'wealt

6375 Context : emerald ---> Target: ['city', 'land', 'oz']
6376 Context : city ---> Target: ['land', 'oz', 'wishes']
6377 Context : land ---> Target: ['oz', 'wishes', 'marry']
6378 Context : oz ---> Target: ['wishes', 'marry', 'daughter']
6379 Context : wishes ---> Target: ['marry', 'daughter', 'princess']
6380 Context : marry ---> Target: ['daughter', 'princess', 'gloria']
6381 Context : daughter ---> Target: ['princess', 'gloria', 'vivian']
6382 Context : princess ---> Target: ['gloria', 'vivian', 'reed']
6383 Context : gloria ---> Target: ['vivian', 'reed', 'old']
6384 Context : vivian ---> Target: ['reed', 'old', 'courtier']
6385 Context : reed ---> Target: ['old', 'courtier', 'named']
6386 Context : old ---> Target: ['courtier', 'named', 'googly']
6387 Context : courtier ---> Target: ['named', 'googly', 'goo']
6388 Context : named ---> Target: ['googly', 'goo', 'arthur']
6389 Context : googly ---> Target: ['goo', 'arthur', 'smollett']
6390 Context : goo ---> Target: ['arthur', 'sm

6575 Context : throughout ---> Target: ['scarecrow', 'makes', 'asides']
6576 Context : scarecrow ---> Target: ['makes', 'asides', 'camera']
6577 Context : makes ---> Target: ['asides', 'camera', 'mostly']
6578 Context : asides ---> Target: ['camera', 'mostly', 'without']
6579 Context : camera ---> Target: ['mostly', 'without', 'intertitles']
6580 Context : mostly ---> Target: ['without', 'intertitles', 'another']
6581 Context : without ---> Target: ['intertitles', 'another', 'point']
6582 Context : intertitles ---> Target: ['another', 'point', 'frozen']
6583 Context : another ---> Target: ['point', 'frozen', 'gloria']
6584 Context : point ---> Target: ['frozen', 'gloria', 'even']
6585 Context : frozen ---> Target: ['gloria', 'even', 'makes']
6586 Context : gloria ---> Target: ['even', 'makes', 'malevolent']
6587 Context : even ---> Target: ['makes', 'malevolent', 'stare']
6588 Context : makes ---> Target: ['malevolent', 'stare', 'directly']
6589 Context : malevolent ---> Target: ['star

6874 Context : also ---> Target: ['known', 'commonly', 'laughing']
6875 Context : known ---> Target: ['commonly', 'laughing', 'gas']
6876 Context : commonly ---> Target: ['laughing', 'gas', 'due']
6877 Context : laughing ---> Target: ['gas', 'due', 'effects']
6878 Context : gas ---> Target: ['due', 'effects', 'prior']
6879 Context : due ---> Target: ['effects', 'prior', 'unconsciousness']
6880 Context : effects ---> Target: ['prior', 'unconsciousness', 'man']
6881 Context : prior ---> Target: ['unconsciousness', 'man', 'unconscious']
6882 Context : unconsciousness ---> Target: ['man', 'unconscious', 'pulls']
6883 Context : man ---> Target: ['unconscious', 'pulls', 'tooth']
6884 Context : unconscious ---> Target: ['pulls', 'tooth', 'get']
6885 Context : pulls ---> Target: ['tooth', 'get', 'wake']
6886 Context : tooth ---> Target: ['get', 'wake', 'calls']
6887 Context : get ---> Target: ['wake', 'calls', 'charlie']
6888 Context : wake ---> Target: ['calls', 'charlie', 'arrives']
6889 Con

7274 Context : couple ---> Target: ['leave', 'mabel', 'heads']
7275 Context : leave ---> Target: ['mabel', 'heads', 'home']
7276 Context : mabel ---> Target: ['heads', 'home', 'stops']
7277 Context : heads ---> Target: ['home', 'stops', 'sporting']
7278 Context : home ---> Target: ['stops', 'sporting', 'goods']
7279 Context : stops ---> Target: ['sporting', 'goods', 'store']
7280 Context : sporting ---> Target: ['goods', 'store', 'orders']
7281 Context : goods ---> Target: ['store', 'orders', 'man']
7282 Context : store ---> Target: ['orders', 'man', 'shaped']
7283 Context : orders ---> Target: ['man', 'shaped', 'punch']
7284 Context : man ---> Target: ['shaped', 'punch', 'bag']
7285 Context : shaped ---> Target: ['punch', 'bag', 'delivered']
7286 Context : punch ---> Target: ['bag', 'delivered', 'whilst']
7287 Context : bag ---> Target: ['delivered', 'whilst', 'pyjamas']
7288 Context : delivered ---> Target: ['whilst', 'pyjamas', 'wraps']
7289 Context : whilst ---> Target: ['pyjamas',

7574 Context : death ---> Target: ['young', 'sister', 'caught']
7575 Context : young ---> Target: ['sister', 'caught', 'fishing']
7576 Context : sister ---> Target: ['caught', 'fishing', 'net']
7577 Context : caught ---> Target: ['fishing', 'net', 'however']
7578 Context : fishing ---> Target: ['net', 'however', 'falls']
7579 Context : net ---> Target: ['however', 'falls', 'love']
7580 Context : however ---> Target: ['falls', 'love', 'king']
7581 Context : falls ---> Target: ['love', 'king', 'man']
7582 Context : love ---> Target: ['king', 'man', 'holds']
7583 Context : king ---> Target: ['man', 'holds', 'responsible']
7584 Context : hero ---> Target: ['janitor', 'played', 'chaplin']
7585 Context : janitor ---> Target: ['played', 'chaplin', 'fired']
7586 Context : played ---> Target: ['chaplin', 'fired', 'work']
7587 Context : chaplin ---> Target: ['fired', 'work', 'accidentally']
7588 Context : fired ---> Target: ['work', 'accidentally', 'knocking']
7589 Context : work ---> Target: ['

7874 Context : inebriated ---> Target: ['neighbor', 'arbuckle', 'goes']
7875 Context : neighbor ---> Target: ['arbuckle', 'goes', 'home']
7876 Context : arbuckle ---> Target: ['goes', 'home', 'cold']
7877 Context : goes ---> Target: ['home', 'cold', 'reception']
7878 Context : home ---> Target: ['cold', 'reception', 'wife']
7879 Context : cold ---> Target: ['reception', 'wife', 'first']
7880 Context : reception ---> Target: ['wife', 'first', 'couple']
7881 Context : wife ---> Target: ['first', 'couple', 'hear']
7882 Context : first ---> Target: ['couple', 'hear', 'physical']
7883 Context : couple ---> Target: ['hear', 'physical', 'altercation']
7884 Context : hear ---> Target: ['physical', 'altercation', 'across']
7885 Context : physical ---> Target: ['altercation', 'across', 'hall']
7886 Context : altercation ---> Target: ['across', 'hall', 'second']
7887 Context : across ---> Target: ['hall', 'second', 'man']
7888 Context : hall ---> Target: ['second', 'man', 'starts']
7889 Context :

8174 Context : hand ---> Target: ['asking', 'making', 'sad']
8175 Context : asking ---> Target: ['making', 'sad', 'child']
8176 Context : making ---> Target: ['sad', 'child', 'reminds']
8177 Context : sad ---> Target: ['child', 'reminds', 'lost']
8178 Context : child ---> Target: ['reminds', 'lost', 'little']
8179 Context : reminds ---> Target: ['lost', 'little', 'girl']
8180 Context : lost ---> Target: ['little', 'girl', 'times']
8181 Context : little ---> Target: ['girl', 'times', 'past']
8182 Context : girl ---> Target: ['times', 'past', 'begin']
8183 Context : times ---> Target: ['past', 'begin', 'flow']
8184 Context : past ---> Target: ['begin', 'flow', 'memory']
8185 Context : begin ---> Target: ['flow', 'memory', 'happy']
8186 Context : flow ---> Target: ['memory', 'happy', 'life']
8187 Context : memory ---> Target: ['happy', 'life', 'loving']
8188 Context : happy ---> Target: ['life', 'loving', 'wife']
8189 Context : life ---> Target: ['loving', 'wife', 'baby']
8190 Context : l

8474 Context : money ---> Target: ['gets', 'tillie', 'drunk']
8475 Context : gets ---> Target: ['tillie', 'drunk', 'restaurant']
8476 Context : tillie ---> Target: ['drunk', 'restaurant', 'asks']
8477 Context : drunk ---> Target: ['restaurant', 'asks', 'let']
8478 Context : restaurant ---> Target: ['asks', 'let', 'hold']
8479 Context : asks ---> Target: ['let', 'hold', 'pocketbook']
8480 Context : let ---> Target: ['hold', 'pocketbook', 'since']
8481 Context : hold ---> Target: ['pocketbook', 'since', 'drunk']
8482 Context : pocketbook ---> Target: ['since', 'drunk', 'agrees']
8483 Context : since ---> Target: ['drunk', 'agrees', 'escapes']
8484 Context : drunk ---> Target: ['agrees', 'escapes', 'old']
8485 Context : agrees ---> Target: ['escapes', 'old', 'girlfriend']
8486 Context : escapes ---> Target: ['old', 'girlfriend', 'money']
8487 Context : old ---> Target: ['girlfriend', 'money', 'later']
8488 Context : girlfriend ---> Target: ['money', 'later', 'day']
8489 Context : money --

8774 Context : volcano ---> Target: ['erupts', 'village', 'destroyed']
8775 Context : erupts ---> Target: ['village', 'destroyed', 'takeo']
8776 Context : village ---> Target: ['destroyed', 'takeo', 'dies']
8777 Context : destroyed ---> Target: ['takeo', 'dies', 'avalanche']
8778 Context : takeo ---> Target: ['dies', 'avalanche', 'tom']
8779 Context : dies ---> Target: ['avalanche', 'tom', 'wilson']
8780 Context : avalanche ---> Target: ['tom', 'wilson', 'toya']
8781 Context : tom ---> Target: ['wilson', 'toya', 'san']
8782 Context : wilson ---> Target: ['toya', 'san', 'survive']
8783 Context : toya ---> Target: ['san', 'survive', 'taken']
8784 Context : san ---> Target: ['survive', 'taken', 'away']
8785 Context : survive ---> Target: ['taken', 'away', 'destroyed']
8786 Context : taken ---> Target: ['away', 'destroyed', 'village']
8787 Context : away ---> Target: ['destroyed', 'village', 'united']
8788 Context : destroyed ---> Target: ['village', 'united', 'states']
8789 Context : vill

9174 Context : trapped ---> Target: ['precipice', 'flora', 'warns']
9175 Context : precipice ---> Target: ['flora', 'warns', 'gus']
9176 Context : flora ---> Target: ['warns', 'gus', 'jump']
9177 Context : warns ---> Target: ['gus', 'jump', 'comes']
9178 Context : gus ---> Target: ['jump', 'comes', 'closer']
9179 Context : jump ---> Target: ['comes', 'closer', 'leaps']
9180 Context : comes ---> Target: ['closer', 'leaps', 'death']
9181 Context : closer ---> Target: ['leaps', 'death', 'run']
9182 Context : leaps ---> Target: ['death', 'run', 'forest']
9183 Context : death ---> Target: ['run', 'forest', 'looking']
9184 Context : run ---> Target: ['forest', 'looking', 'ben']
9185 Context : forest ---> Target: ['looking', 'ben', 'seen']
9186 Context : looking ---> Target: ['ben', 'seen', 'jump']
9187 Context : ben ---> Target: ['seen', 'jump', 'holds']
9188 Context : seen ---> Target: ['jump', 'holds', 'dies']
9189 Context : jump ---> Target: ['holds', 'dies', 'carries']
9190 Context : hol

9574 Context : shares ---> Target: ['national', 'origin', 'turkish']
9575 Context : national ---> Target: ['origin', 'turkish', 'army']
9576 Context : origin ---> Target: ['turkish', 'army', 'driven']
9577 Context : turkish ---> Target: ['army', 'driven', 'village']
9578 Context : army ---> Target: ['driven', 'village', 'hassan']
9579 Context : driven ---> Target: ['village', 'hassan', 'returns']
9580 Context : village ---> Target: ['hassan', 'returns', 'home']
9581 Context : hassan ---> Target: ['returns', 'home', 'faced']
9582 Context : returns ---> Target: ['home', 'faced', 'grim']
9583 Context : home ---> Target: ['faced', 'grim', 'reality']
9584 Context : faced ---> Target: ['grim', 'reality', 'stripped']
9585 Context : grim ---> Target: ['reality', 'stripped', 'title']
9586 Context : reality ---> Target: ['stripped', 'title', 'land']
9587 Context : stripped ---> Target: ['title', 'land', 'taken']
9588 Context : title ---> Target: ['land', 'taken', 'banished']
9589 Context : land 

9874 Context : pretend ---> Target: ['find', 'gold', 'retreat']
9875 Context : find ---> Target: ['gold', 'retreat', 'company']
9876 Context : gold ---> Target: ['retreat', 'company', 'takes']
9877 Context : retreat ---> Target: ['company', 'takes', 'advantage']
9878 Context : company ---> Target: ['takes', 'advantage', 'things']
9879 Context : takes ---> Target: ['advantage', 'things', 'go']
9880 Context : advantage ---> Target: ['things', 'go', 'planned']
9881 Context : based ---> Target: ['summary', 'film', 'magazine']
9882 Context : summary ---> Target: ['film', 'magazine', 'enoch']
9883 Context : film ---> Target: ['magazine', 'enoch', 'annie']
9884 Context : magazine ---> Target: ['enoch', 'annie', 'walter']
9885 Context : enoch ---> Target: ['annie', 'walter', 'grow']
9886 Context : annie ---> Target: ['walter', 'grow', 'friends']
9887 Context : walter ---> Target: ['grow', 'friends', 'later']
9888 Context : grow ---> Target: ['friends', 'later', 'annie']
9889 Context : friends 

10180 Context : jail ---> Target: ['must', 'get', 'de']
10181 Context : must ---> Target: ['get', 'de', 'milk']
10182 Context : get ---> Target: ['de', 'milk', 'babee']
10183 Context : de ---> Target: ['milk', 'babee', 'die']
10184 Context : milk ---> Target: ['babee', 'die', 'corrigan']
10185 Context : babee ---> Target: ['die', 'corrigan', 'rebuffs']
10186 Context : die ---> Target: ['corrigan', 'rebuffs', 'beppo']
10187 Context : corrigan ---> Target: ['rebuffs', 'beppo', 'beppo']
10188 Context : rebuffs ---> Target: ['beppo', 'beppo', 'baby']
10189 Context : beppo ---> Target: ['beppo', 'baby', 'dies']
10190 Context : beppo ---> Target: ['baby', 'dies', 'beppo']
10191 Context : baby ---> Target: ['dies', 'beppo', 'five']
10192 Context : dies ---> Target: ['beppo', 'five', 'days']
10193 Context : beppo ---> Target: ['five', 'days', 'jail']
10194 Context : five ---> Target: ['days', 'jail', 'beppo']
10195 Context : days ---> Target: ['jail', 'beppo', 'released']
10196 Context : jail 

10473 Context : spending ---> Target: ['habits', 'allan', 'causes']
10474 Context : habits ---> Target: ['allan', 'causes', 'quite']
10475 Context : allan ---> Target: ['causes', 'quite', 'scene']
10476 Context : causes ---> Target: ['quite', 'scene', 'despite']
10477 Context : quite ---> Target: ['scene', 'despite', 'wife']
10478 Context : scene ---> Target: ['despite', 'wife', 'attempts']
10479 Context : despite ---> Target: ['wife', 'attempts', 'calm']
10480 Context : wife ---> Target: ['attempts', 'calm', 'poe']
10481 Context : attempts ---> Target: ['calm', 'poe', 'asked']
10482 Context : calm ---> Target: ['poe', 'asked', 'leave']
10483 Context : poe ---> Target: ['asked', 'leave', 'allan']
10484 Context : asked ---> Target: ['leave', 'allan', 'family']
10485 Context : leave ---> Target: ['allan', 'family', 'virginia']
10486 Context : allan ---> Target: ['family', 'virginia', 'offers']
10487 Context : family ---> Target: ['virginia', 'offers', 'come']
10488 Context : virginia ---

10787 Context : chaplin ---> Target: ['finds', 'girl', 'dreams']
10788 Context : finds ---> Target: ['girl', 'dreams', 'works']
10789 Context : girl ---> Target: ['dreams', 'works', 'family']
10790 Context : dreams ---> Target: ['works', 'family', 'farm']
10791 Context : works ---> Target: ['family', 'farm', 'helps']
10792 Context : family ---> Target: ['farm', 'helps', 'defend']
10793 Context : farm ---> Target: ['helps', 'defend', 'farm']
10794 Context : helps ---> Target: ['defend', 'farm', 'criminals']
10795 Context : defend ---> Target: ['farm', 'criminals', 'seems']
10796 Context : farm ---> Target: ['criminals', 'seems', 'well']
10797 Context : criminals ---> Target: ['seems', 'well', 'discovers']
10798 Context : seems ---> Target: ['well', 'discovers', 'girl']
10799 Context : well ---> Target: ['discovers', 'girl', 'dreams']
10800 Context : discovers ---> Target: ['girl', 'dreams', 'already']
10801 Context : girl ---> Target: ['dreams', 'already', 'boyfriend']
10802 Context : d

11173 Context : emily ---> Target: ['grows', 'resentful', 'dedication']
11174 Context : grows ---> Target: ['resentful', 'dedication', 'duty']
11175 Context : resentful ---> Target: ['dedication', 'duty', 'curtailed']
11176 Context : dedication ---> Target: ['duty', 'curtailed', 'much']
11177 Context : duty ---> Target: ['curtailed', 'much', 'social']
11178 Context : curtailed ---> Target: ['much', 'social', 'life']
11179 Context : much ---> Target: ['social', 'life', 'starts']
11180 Context : social ---> Target: ['life', 'starts', 'spending']
11181 Context : life ---> Target: ['starts', 'spending', 'time']
11182 Context : starts ---> Target: ['spending', 'time', 'benson']
11183 Context : spending ---> Target: ['time', 'benson', 'heath']
11184 Context : time ---> Target: ['benson', 'heath', 'heath']
11185 Context : benson ---> Target: ['heath', 'heath', 'real']
11186 Context : heath ---> Target: ['heath', 'real', 'job']
11187 Context : heath ---> Target: ['real', 'job', 'attracted']
11

11573 Context : abandoning ---> Target: ['mother', 'father', 'dies']
11574 Context : mother ---> Target: ['father', 'dies', 'jason']
11575 Context : father ---> Target: ['dies', 'jason', 'turns']
11576 Context : dies ---> Target: ['jason', 'turns', 'desire']
11577 Context : jason ---> Target: ['turns', 'desire', 'revenge']
11578 Context : turns ---> Target: ['desire', 'revenge', 'sunlocks']
11579 Context : desire ---> Target: ['revenge', 'sunlocks', 'father']
11580 Context : revenge ---> Target: ['sunlocks', 'father', 'son']
11581 Context : sunlocks ---> Target: ['father', 'son', 'another']
11582 Context : father ---> Target: ['son', 'another', 'wife']
11583 Context : son ---> Target: ['another', 'wife', 'sunlocks']
11584 Context : another ---> Target: ['wife', 'sunlocks', 'jason']
11585 Context : wife ---> Target: ['sunlocks', 'jason', 'love']
11586 Context : sunlocks ---> Target: ['jason', 'love', 'greeba']
11587 Context : jason ---> Target: ['love', 'greeba', 'daughter']
11588 Conte

11973 Context : meanwhile ---> Target: ['father', 'deliberately', 'sets']
11974 Context : father ---> Target: ['deliberately', 'sets', 'fire']
11975 Context : deliberately ---> Target: ['sets', 'fire', 'basement']
11976 Context : sets ---> Target: ['fire', 'basement', 'house']
11977 Context : fire ---> Target: ['basement', 'house', 'without']
11978 Context : basement ---> Target: ['house', 'without', 'realizing']
11979 Context : house ---> Target: ['without', 'realizing', 'daughter']
11980 Context : without ---> Target: ['realizing', 'daughter', 'still']
11981 Context : realizing ---> Target: ['daughter', 'still', 'inside']
11982 Context : daughter ---> Target: ['still', 'inside', 'house']
11983 Context : still ---> Target: ['inside', 'house', 'upper']
11984 Context : inside ---> Target: ['house', 'upper', 'floor']
11985 Context : house ---> Target: ['upper', 'floor', 'upon']
11986 Context : upper ---> Target: ['floor', 'upon', 'knowing']
11987 Context : floor ---> Target: ['upon', 'kn

12373 Context : afterwards ---> Target: ['uncle', 'peter', 'drives']
12374 Context : uncle ---> Target: ['peter', 'drives', 'port']
12375 Context : peter ---> Target: ['drives', 'port', 'new']
12376 Context : drives ---> Target: ['port', 'new', 'york']
12377 Context : port ---> Target: ['new', 'york', 'pick']
12378 Context : new ---> Target: ['york', 'pick', 'injured']
12379 Context : york ---> Target: ['pick', 'injured', 'automobile']
12380 Context : pick ---> Target: ['injured', 'automobile', 'accident']
12381 Context : injured ---> Target: ['automobile', 'accident', 'taken']
12382 Context : automobile ---> Target: ['accident', 'taken', 'hospital']
12383 Context : accident ---> Target: ['taken', 'hospital', 'lies']
12384 Context : taken ---> Target: ['hospital', 'lies', 'unidentified']
12385 Context : hospital ---> Target: ['lies', 'unidentified', 'uncle']
12386 Context : lies ---> Target: ['unidentified', 'uncle', 'peter']
12387 Context : unidentified ---> Target: ['uncle', 'peter',

12773 Context : gregory ---> Target: ['savors', 'latest', 'acquisition']
12774 Context : savors ---> Target: ['latest', 'acquisition', 'secret']
12775 Context : latest ---> Target: ['acquisition', 'secret', 'henchman']
12776 Context : acquisition ---> Target: ['secret', 'henchman', 'suggests']
12777 Context : secret ---> Target: ['henchman', 'suggests', 'getting']
12778 Context : henchman ---> Target: ['suggests', 'getting', 'rid']
12779 Context : suggests ---> Target: ['getting', 'rid', 'melville']
12780 Context : getting ---> Target: ['rid', 'melville', 'gregory']
12781 Context : rid ---> Target: ['melville', 'gregory', 'turns']
12782 Context : melville ---> Target: ['gregory', 'turns', 'found']
12783 Context : gregory ---> Target: ['turns', 'found', 'worthy']
12784 Context : turns ---> Target: ['found', 'worthy', 'foe']
12785 Context : found ---> Target: ['worthy', 'foe', 'something']
12786 Context : worthy ---> Target: ['foe', 'something', 'crude']
12787 Context : foe ---> Target: 

13173 Context : millennia ---> Target: ['later', 'nadia', 'agnes']
13174 Context : later ---> Target: ['nadia', 'agnes', 'daughter']
13175 Context : nadia ---> Target: ['agnes', 'daughter', 'millionaire']
13176 Context : agnes ---> Target: ['daughter', 'millionaire', 'lucian']
13177 Context : daughter ---> Target: ['millionaire', 'lucian', 'arthur']
13178 Context : millionaire ---> Target: ['lucian', 'arthur', 'wall']
13179 Context : lucian ---> Target: ['arthur', 'wall', 'street']
13180 Context : arthur ---> Target: ['wall', 'street', 'broker']
13181 Context : mary ---> Target: ['beresford', 'boland', 'wife']
13182 Context : beresford ---> Target: ['boland', 'wife', 'unambitious']
13183 Context : boland ---> Target: ['wife', 'unambitious', 'law']
13184 Context : wife ---> Target: ['unambitious', 'law', 'clerk']
13185 Context : unambitious ---> Target: ['law', 'clerk', 'al']
13186 Context : law ---> Target: ['clerk', 'al', 'beresford']
13187 Context : clerk ---> Target: ['al', 'beresfo

13573 Context : jimmie ---> Target: ['tries', 'desperately', 'foil']
13574 Context : tries ---> Target: ['desperately', 'foil', 'attempt']
13575 Context : desperately ---> Target: ['foil', 'attempt', 'robbery']
13576 Context : foil ---> Target: ['attempt', 'robbery', 'wounded']
13577 Context : attempt ---> Target: ['robbery', 'wounded', 'arrested']
13578 Context : robbery ---> Target: ['wounded', 'arrested', 'sheriff']
13579 Context : wounded ---> Target: ['arrested', 'sheriff', 'william']
13580 Context : arrested ---> Target: ['sheriff', 'william', 'h']
13581 Context : sheriff ---> Target: ['william', 'h', 'brown']
13582 Context : william ---> Target: ['h', 'brown', 'robbery']
13583 Context : h ---> Target: ['brown', 'robbery', 'suspect']
13584 Context : brown ---> Target: ['robbery', 'suspect', 'jimmie']
13585 Context : robbery ---> Target: ['suspect', 'jimmie', 'escapes']
13586 Context : suspect ---> Target: ['jimmie', 'escapes', 'jail']
13587 Context : jimmie ---> Target: ['escapes

13972 Context : summary ---> Target: ['reconstructed', 'description', 'contemporary']
13973 Context : reconstructed ---> Target: ['description', 'contemporary', 'film']
13974 Context : description ---> Target: ['contemporary', 'film', 'magazine']
13975 Context : contemporary ---> Target: ['film', 'magazine', 'cleopatra']
13976 Context : film ---> Target: ['magazine', 'cleopatra', 'bara']
13977 Context : magazine ---> Target: ['cleopatra', 'bara', 'siren']
13978 Context : cleopatra ---> Target: ['bara', 'siren', 'egypt']
13979 Context : bara ---> Target: ['siren', 'egypt', 'clever']
13980 Context : siren ---> Target: ['egypt', 'clever', 'ruse']
13981 Context : egypt ---> Target: ['clever', 'ruse', 'reaches']
13982 Context : clever ---> Target: ['ruse', 'reaches', 'caesar']
13983 Context : ruse ---> Target: ['reaches', 'caesar', 'leiber']
13984 Context : reaches ---> Target: ['caesar', 'leiber', 'falls']
13985 Context : caesar ---> Target: ['leiber', 'falls', 'victim']
13986 Context : le

14372 Context : caught ---> Target: ['unexpected', 'storm', 'date']
14373 Context : unexpected ---> Target: ['storm', 'date', 'course']
14374 Context : storm ---> Target: ['date', 'course', 'gets']
14375 Context : date ---> Target: ['course', 'gets', 'pregnant']
14376 Context : course ---> Target: ['gets', 'pregnant', 'mom']
14377 Context : gets ---> Target: ['pregnant', 'mom', 'realize']
14378 Context : pregnant ---> Target: ['mom', 'realize', 'lillian']
14379 Context : mom ---> Target: ['realize', 'lillian', 'night']
14380 Context : realize ---> Target: ['lillian', 'night', 'due']
14381 Context : lillian ---> Target: ['night', 'due', 'gambling']
14382 Context : night ---> Target: ['due', 'gambling', 'addiction']
14383 Context : due ---> Target: ['gambling', 'addiction', 'young']
14384 Context : gambling ---> Target: ['addiction', 'young', 'man']
14385 Context : addiction ---> Target: ['young', 'man', 'later']
14386 Context : young ---> Target: ['man', 'later', 'starts']
14387 Context

14672 Context : orphaned ---> Target: ['children', 'taken', 'uncle']
14673 Context : children ---> Target: ['taken', 'uncle', 'mayor']
14674 Context : taken ---> Target: ['uncle', 'mayor', 'hoadley']
14675 Context : uncle ---> Target: ['mayor', 'hoadley', 'john']
14676 Context : mayor ---> Target: ['hoadley', 'john', 'robertson']
14677 Context : hoadley ---> Target: ['john', 'robertson', 'hoadley']
14678 Context : john ---> Target: ['robertson', 'hoadley', 'despises']
14679 Context : robertson ---> Target: ['hoadley', 'despises', 'children']
14680 Context : hoadley ---> Target: ['despises', 'children', 'taken']
14681 Context : despises ---> Target: ['children', 'taken', 'good']
14682 Context : children ---> Target: ['taken', 'good', 'publicity']
14683 Context : taken ---> Target: ['good', 'publicity', 'upcoming']
14684 Context : good ---> Target: ['publicity', 'upcoming', 'election']
14685 Context : publicity ---> Target: ['upcoming', 'election', 'wife']
14686 Context : upcoming ---> T

15072 Context : alice ---> Target: ['discovering', 'happened', 'gets']
15073 Context : discovering ---> Target: ['happened', 'gets', 'bicycle']
15074 Context : happened ---> Target: ['gets', 'bicycle', 'heads']
15075 Context : gets ---> Target: ['bicycle', 'heads', 'justice']
15076 Context : bicycle ---> Target: ['heads', 'justice', 'peace']
15077 Context : heads ---> Target: ['justice', 'peace', 'office']
15078 Context : justice ---> Target: ['peace', 'office', 'well']
15079 Context : peace ---> Target: ['office', 'well', 'arrives']
15080 Context : office ---> Target: ['well', 'arrives', 'time']
15081 Context : well ---> Target: ['arrives', 'time', 'stop']
15082 Context : arrives ---> Target: ['time', 'stop', 'clerk']
15083 Context : time ---> Target: ['stop', 'clerk', 'marrying']
15084 Context : stop ---> Target: ['clerk', 'marrying', 'delivery']
15085 Context : clerk ---> Target: ['marrying', 'delivery', 'boy']
15086 Context : marrying ---> Target: ['delivery', 'boy', 'arbuckle']
15

15472 Context : goes ---> Target: ['montreal', 'fulfill', 'ancient']
15473 Context : montreal ---> Target: ['fulfill', 'ancient', 'pact']
15474 Context : fulfill ---> Target: ['ancient', 'pact', 'finds']
15475 Context : ancient ---> Target: ['pact', 'finds', 'happiness']
15476 Context : karl ---> Target: ['von', 'austreim', 'jack']
15477 Context : von ---> Target: ['austreim', 'jack', 'holt']
15478 Context : austreim ---> Target: ['jack', 'holt', 'lives']
15479 Context : jack ---> Target: ['holt', 'lives', 'america']
15480 Context : holt ---> Target: ['lives', 'america', 'german']
15481 Context : lives ---> Target: ['america', 'german', 'father']
15482 Context : america ---> Target: ['german', 'father', 'american']
15483 Context : german ---> Target: ['father', 'american', 'mother']
15484 Context : father ---> Target: ['american', 'mother', 'notices']
15485 Context : american ---> Target: ['mother', 'notices', 'young']
15486 Context : mother ---> Target: ['notices', 'young', 'lady']
15

15872 Context : precedes ---> Target: ['girls', 'fascinated', 'tales']
15873 Context : girls ---> Target: ['fascinated', 'tales', 'life']
15874 Context : fascinated ---> Target: ['tales', 'life', 'india']
15875 Context : tales ---> Target: ['life', 'india', 'girls']
15876 Context : life ---> Target: ['india', 'girls', 'sneak']
15877 Context : india ---> Target: ['girls', 'sneak', 'sarah']
15878 Context : girls ---> Target: ['sneak', 'sarah', 'room']
15879 Context : sneak ---> Target: ['sarah', 'room', 'night']
15880 Context : sarah ---> Target: ['room', 'night', 'listen']
15881 Context : room ---> Target: ['night', 'listen', 'stories']
15882 Context : night ---> Target: ['listen', 'stories', 'one']
15883 Context : listen ---> Target: ['stories', 'one', 'night']
15884 Context : stories ---> Target: ['one', 'night', 'tells']
15885 Context : one ---> Target: ['night', 'tells', 'ali']
15886 Context : night ---> Target: ['tells', 'ali', 'baba']
15887 Context : tells ---> Target: ['ali', 'ba

16272 Context : kansas ---> Target: ['mirroring', 'scenes', 'artagnan']
16273 Context : mirroring ---> Target: ['scenes', 'artagnan', 'astride']
16274 Context : scenes ---> Target: ['artagnan', 'astride', 'somewhat']
16275 Context : artagnan ---> Target: ['astride', 'somewhat', 'less']
16276 Context : astride ---> Target: ['somewhat', 'less', 'noble']
16277 Context : somewhat ---> Target: ['less', 'noble', 'steed']
16278 Context : less ---> Target: ['noble', 'steed', 'present']
16279 Context : noble ---> Target: ['steed', 'present', 'father']
16280 Context : steed ---> Target: ['present', 'father', 'departure']
16281 Context : present ---> Target: ['father', 'departure', 'home']
16282 Context : father ---> Target: ['departure', 'home', 'ned']
16283 Context : departure ---> Target: ['home', 'ned', 'father']
16284 Context : home ---> Target: ['ned', 'father', 'gives']
16285 Context : ned ---> Target: ['father', 'gives', 'modern']
16286 Context : father ---> Target: ['gives', 'modern', 'e

16672 Context : tale ---> Target: ['forest', 'conditions', 'worsens']
16673 Context : forest ---> Target: ['conditions', 'worsens', 'death']
16674 Context : conditions ---> Target: ['worsens', 'death', 'tries']
16675 Context : worsens ---> Target: ['death', 'tries', 'lure']
16676 Context : death ---> Target: ['tries', 'lure', 'eternal']
16677 Context : tries ---> Target: ['lure', 'eternal', 'rest']
16678 Context : lure ---> Target: ['eternal', 'rest', 'life']
16679 Context : eternal ---> Target: ['rest', 'life', 'also']
16680 Context : rest ---> Target: ['life', 'also', 'appears']
16681 Context : life ---> Target: ['also', 'appears', 'finally']
16682 Context : also ---> Target: ['appears', 'finally', 'wins']
16683 Context : described ---> Target: ['film', 'magazine', 'alexis']
16684 Context : film ---> Target: ['magazine', 'alexis', 'caesar']
16685 Context : magazine ---> Target: ['alexis', 'caesar', 'napoleon']
16686 Context : alexis ---> Target: ['caesar', 'napoleon', 'brown']
16687 

17072 Context : house ---> Target: ['time', 'unintentionally', 'stop']
17073 Context : time ---> Target: ['unintentionally', 'stop', 'fleeing']
17074 Context : unintentionally ---> Target: ['stop', 'fleeing', 'thieves']
17075 Context : stop ---> Target: ['fleeing', 'thieves', 'bumping']
17076 Context : fleeing ---> Target: ['thieves', 'bumping', 'mr']
17077 Context : thieves ---> Target: ['bumping', 'mr', 'rough']
17078 Context : bumping ---> Target: ['mr', 'rough', 'takes']
17079 Context : mr ---> Target: ['rough', 'takes', 'back']
17080 Context : rough ---> Target: ['takes', 'back', 'necklace']
17081 Context : takes ---> Target: ['back', 'necklace', 'thieves']
17082 Context : back ---> Target: ['necklace', 'thieves', 'taken']
17083 Context : necklace ---> Target: ['thieves', 'taken', 'jail']
17084 Context : described ---> Target: ['film', 'magazine', 'review']
17085 Context : film ---> Target: ['magazine', 'review', 'sebastian']
17086 Context : magazine ---> Target: ['review', 'sebas

17472 Context : alvarado ---> Target: ['comes', 'wooing', 'last']
17473 Context : comes ---> Target: ['wooing', 'last', 'aztecs']
17474 Context : wooing ---> Target: ['last', 'aztecs', 'wins']
17475 Context : described ---> Target: ['film', 'magazine', 'susan']
17476 Context : film ---> Target: ['magazine', 'susan', 'sweeney']
17477 Context : magazine ---> Target: ['susan', 'sweeney', 'marsh']
17478 Context : susan ---> Target: ['sweeney', 'marsh', 'employed']
17479 Context : sweeney ---> Target: ['marsh', 'employed', 'doll']
17480 Context : marsh ---> Target: ['employed', 'doll', 'factory']
17481 Context : employed ---> Target: ['doll', 'factory', 'learns']
17482 Context : doll ---> Target: ['factory', 'learns', 'inherited']
17483 Context : factory ---> Target: ['learns', 'inherited', 'hotel']
17484 Context : learns ---> Target: ['inherited', 'hotel', 'small']
17485 Context : inherited ---> Target: ['hotel', 'small', 'town']
17486 Context : hotel ---> Target: ['small', 'town', 'adiron

17771 Context : affair ---> Target: ['denton', 'follows', 'estrella']
17772 Context : denton ---> Target: ['follows', 'estrella', 'room']
17773 Context : follows ---> Target: ['estrella', 'room', 'benham']
17774 Context : estrella ---> Target: ['room', 'benham', 'catches']
17775 Context : room ---> Target: ['benham', 'catches', 'misunderstands']
17776 Context : benham ---> Target: ['catches', 'misunderstands', 'sees']
17777 Context : catches ---> Target: ['misunderstands', 'sees', 'denton']
17778 Context : misunderstands ---> Target: ['sees', 'denton', 'honorably']
17779 Context : sees ---> Target: ['denton', 'honorably', 'keeps']
17780 Context : denton ---> Target: ['honorably', 'keeps', 'estrella']
17781 Context : honorably ---> Target: ['keeps', 'estrella', 'secret']
17782 Context : keeps ---> Target: ['estrella', 'secret', 'consequence']
17783 Context : estrella ---> Target: ['secret', 'consequence', 'must']
17784 Context : secret ---> Target: ['consequence', 'must', 'resign']
1778

18171 Context : berlingot ---> Target: ['transforming', 'true', 'beautiful']
18172 Context : transforming ---> Target: ['true', 'beautiful', 'appearance']
18173 Context : true ---> Target: ['beautiful', 'appearance', 'insists']
18174 Context : beautiful ---> Target: ['appearance', 'insists', 'children']
18175 Context : appearance ---> Target: ['insists', 'children', 'search']
18176 Context : insists ---> Target: ['children', 'search', 'bluebird']
18177 Context : children ---> Target: ['search', 'bluebird', 'happiness']
18178 Context : search ---> Target: ['bluebird', 'happiness', 'gives']
18179 Context : bluebird ---> Target: ['happiness', 'gives', 'tyltyl']
18180 Context : happiness ---> Target: ['gives', 'tyltyl', 'magical']
18181 Context : gives ---> Target: ['tyltyl', 'magical', 'hat']
18182 Context : tyltyl ---> Target: ['magical', 'hat', 'power']
18183 Context : magical ---> Target: ['hat', 'power', 'show']
18184 Context : hat ---> Target: ['power', 'show', 'insides']
18185 Conte

In [33]:
#reshaping dataset to fit in neural network
r = [X.shape[0],X.shape[2]]
X = X.reshape(r)
Y = Y.reshape(r)

### Model Definition

In [34]:
# create model
model = Sequential()
model.add(Dense(100, input_dim=r[1], init= 'uniform' ,kernel_constraint=nonneg(), activation= 'sigmoid' ))
model.add(Dense(r[1], init= 'uniform',kernel_constraint=nonneg(), activation= 'softmax' ))

model.compile(loss= 'binary_crossentropy' , optimizer= 'adam' , metrics=['accuracy'])

W0829 21:32:45.312567  9500 deprecation_wrapper.py:119] From C:\ProgramData\Anaconda3\lib\site-packages\keras\backend\tensorflow_backend.py:74: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: UserWarning: Update your `Dense` call to the Keras 2 API: `Dense(100, input_dim=5779, kernel_constraint=<keras.con..., activation="sigmoid", kernel_initializer="uniform")`
  This is separate from the ipykernel package so we can avoid doing imports until
W0829 21:32:46.774575  9500 deprecation_wrapper.py:119] From C:\ProgramData\Anaconda3\lib\site-packages\keras\backend\tensorflow_backend.py:517: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

W0829 21:32:46.972735  9500 deprecation_wrapper.py:119] From C:\ProgramData\Anaconda3\lib\site-packages\keras\backend\tensorflow_backend.py:4138: The name tf.random_uniform is deprecated. Please use tf.rando

### Training Word Vectors

In [35]:
train = model.fit(X, Y, nb_epoch=1)

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: UserWarning: The `nb_epoch` argument in `fit` has been renamed `epochs`.
  """Entry point for launching an IPython kernel.
W0829 21:32:48.037503  9500 deprecation_wrapper.py:119] From C:\ProgramData\Anaconda3\lib\site-packages\keras\backend\tensorflow_backend.py:986: The name tf.assign_add is deprecated. Please use tf.compat.v1.assign_add instead.



Epoch 1/1
18360/18360 [==============================] - 33s 2ms/step - loss: 0.0044 - acc: 0.9995


### Word Similarity Analysis
Now we will use the weight matrix of our Neural Network.<br>
We have two weight metrices for two layers. We have taken 100 hidden units in hidden layer. So, for each word we have 100 dimentional vector.
<br><br>
Weights from both layers need to be summed for final vector.
<br><br>
**Here is an important thing:** Word vectors have less dimention than the one-hot encoding matrix. It reduce computation complexity in NLP task. 

In [36]:
weight1, biases = model.layers[0].get_weights()
weight2, biases = model.layers[1].get_weights()
weight = weight1 +weight2.T

In [37]:
print(word_similarity('afternoon','dead',weight,word2idx))

0.7319193


In [38]:
print(most_similiar('man',weight,word2idx))

[('wife', 0.95879227), ('find', 0.9579319), ('woman', 0.9579221), ('girl', 0.9572276), ('young', 0.9562159)]


### Reproducing Past Results 
In this section weights from the first training article to reproduce the past result that has been discussed in the medium article

In [39]:
from sklearn.externals import joblib

In [40]:
filename = 'model.sav'
#joblib.dump(model, filename)

In [41]:
loaded_model = joblib.load(filename)

In [42]:
W1, biases = loaded_model.layers[0].get_weights()
W2, biases = loaded_model.layers[1].get_weights()
W = W1 +W2.T

In [43]:
print(most_similiar('man',W,word2idx))

[('father', 0.95607513), ('young', 0.9553033), ('wife', 0.94956106), ('back', 0.94942164), ('comes', 0.9466744)]


In [44]:
W[0]

array([0.00821561, 0.04129512, 0.00794185, 0.00780243, 0.01642974,
       0.03365516, 0.01118944, 0.03263743, 0.02428928, 0.0077619 ,
       0.00802749, 0.01934635, 0.03983404, 0.00800348, 0.06027178,
       0.03288328, 0.04176749, 0.01901086, 0.017279  , 0.02211751,
       0.05443039, 0.01153891, 0.00776618, 0.06963662, 0.00784243,
       0.00779849, 0.06005738, 0.07359082, 0.03126929, 0.0289063 ,
       0.01402762, 0.06246269, 0.01757432, 0.0672893 , 0.03275475,
       0.02430024, 0.04265555, 0.04689769, 0.01273581, 0.0290504 ,
       0.02346605, 0.00780975, 0.00773496, 0.04264919, 0.01187569,
       0.047987  , 0.06579044, 0.03375261, 0.00777987, 0.02008096,
       0.02719541, 0.0379695 , 0.0289874 , 0.00775713, 0.00772526,
       0.00790034, 0.01972924, 0.03903933, 0.00775233, 0.04250645,
       0.03706887, 0.00781126, 0.00775632, 0.03585426, 0.02283861,
       0.07334395, 0.00779458, 0.01583624, 0.0104772 , 0.0528849 ,
       0.0079683 , 0.0080232 , 0.00776479, 0.05910534, 0.00817

In [45]:
W

array([[ 0.00821561,  0.04129512,  0.00794185, ...,  0.02667897,
         0.00781133,  0.0078567 ],
       [ 0.01905388,  0.04548687,  0.08296685, ...,  0.06120104,
         0.08697856,  0.04136325],
       [ 0.07770883,  0.08835563,  0.04529092, ...,  0.0320814 ,
         0.06857783,  0.04764068],
       ...,
       [ 0.00774618,  0.04764719,  0.00251844, ...,  0.01819674,
         0.07319212,  0.00442693],
       [ 0.03006158,  0.035986  ,  0.01733726, ..., -0.        ,
         0.00018429,  0.01144186],
       [ 0.01860016,  0.02935425,  0.01771664, ...,  0.00601502,
         0.03184192,  0.04288519]], dtype=float32)

In [46]:
len(list((word2idx)))

5779

In [47]:
W[0][0]

0.008215612

In [48]:
X.shape

(18360, 5779)

In [49]:
word2idx

{'bartender': 0,
 'working': 1,
 'saloon': 2,
 'serving': 3,
 'drinks': 4,
 'customers': 5,
 'fills': 6,
 'stereotypically': 7,
 'irish': 8,
 'man': 9,
 'bucket': 10,
 'beer': 11,
 'carrie': 12,
 'nation': 13,
 'followers': 14,
 'burst': 15,
 'inside': 16,
 'assault': 17,
 'pulling': 18,
 'hat': 19,
 'eyes': 20,
 'dumping': 21,
 'head': 22,
 'group': 23,
 'begin': 24,
 'wrecking': 25,
 'bar': 26,
 'smashing': 27,
 'fixtures': 28,
 'mirrors': 29,
 'breaking': 30,
 'cash': 31,
 'register': 32,
 'sprays': 33,
 'seltzer': 34,
 'water': 35,
 'face': 36,
 'policemen': 37,
 'appear': 38,
 'order': 39,
 'everybody': 40,
 'leave': 41,
 'moon': 42,
 'painted': 43,
 'smiling': 44,
 'hangs': 45,
 'park': 46,
 'night': 47,
 'young': 48,
 'couple': 49,
 'walking': 50,
 'past': 51,
 'fence': 52,
 'learn': 53,
 'railing': 54,
 'look': 55,
 'smiles': 56,
 'embrace': 57,
 'smile': 58,
 'gets': 59,
 'bigger': 60,
 'sit': 61,
 'bench': 62,
 'tree': 63,
 'view': 64,
 'blocked': 65,
 'causing': 66,
 'frown'

In [50]:
X


array([[1., 0., 0., ..., 0., 0., 0.],
       [0., 1., 0., ..., 0., 0., 0.],
       [0., 0., 1., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]])

In [51]:
X_context

['bartender',
 'working',
 'saloon',
 'serving',
 'drinks',
 'customers',
 'fills',
 'stereotypically',
 'irish',
 'man',
 'bucket',
 'beer',
 'carrie',
 'nation',
 'followers',
 'burst',
 'inside',
 'assault',
 'irish',
 'man',
 'pulling',
 'hat',
 'eyes',
 'dumping',
 'beer',
 'head',
 'group',
 'begin',
 'wrecking',
 'bar',
 'smashing',
 'fixtures',
 'mirrors',
 'breaking',
 'cash',
 'register',
 'bartender',
 'sprays',
 'seltzer',
 'water',
 'nation',
 'face',
 'group',
 'policemen',
 'appear',
 'moon',
 'painted',
 'smiling',
 'face',
 'hangs',
 'park',
 'night',
 'young',
 'couple',
 'walking',
 'past',
 'fence',
 'learn',
 'railing',
 'look',
 'moon',
 'smiles',
 'embrace',
 'moon',
 'smile',
 'gets',
 'bigger',
 'sit',
 'bench',
 'tree',
 'moon',
 'view',
 'blocked',
 'causing',
 'frown',
 'last',
 'scene',
 'man',
 'fans',
 'woman',
 'hat',
 'moon',
 'left',
 'sky',
 'perched',
 'shoulder',
 'film',
 'minute',
 'long',
 'composed',
 'two',
 'shots',
 'first',
 'girl',
 'sits',